src/hdmf_zarr/zarr_utils.py

```diff
@@ -151,7 +151,8 @@ def dtype(self):
         return self.__dtype
 
     def __getitem__(self, arg):
-        rows = copy(super().__getitem__(arg))
+        rows = list(copy(super().__getitem__(arg)))
+        # breakpoint()
```

Comment:
If I understand what you are saying correctly, the issue appears when ``self.__swap_refs`` is being called because the individual rows are now represented as tuples. I.e, `rows` is either a tuple (if `arg` is an int) or a list of tuples (if `arg` selects multiple rows). If that is the issue, then I think the we could update ``self.__swap_refs`` to return a new instance of the row (rather than updating the existing one). Here is what I think might work: 


```
def __getitem__(self, arg):
        rows = copy(super().__getitem__(arg))
        if np.issubdtype(type(arg), np.integer):
            rows =  self.__swap_refs(rows)
        else:
            rows = [self.__swap_refs(row) for row in rows]
       return rows

def __swap_refs(self, row):
        updated_row = list(row)  # convert tuple to a list so we can update it
        for i in self.__refgetters:
            getref = self.__refgetters[i]
            updated_row[i] = getref(row[i])
        return updated_row   # TODO if we want to keep these as tuples then we could convert them back here 
```